app/assets/javascripts/editor.js

```diff
@@ -23,24 +26,25 @@ $E = {
         return code;
       }
     });
-
   },
   is_editing: function() {
     return ($E.textarea[0].selectionStart == 0 && $E.textarea[0].selectionEnd == 0)
   },
-
   refresh: function() {
-      if($D.selected) {
-          $E.textarea = ($D.selected).find('textarea').eq(0);
-          $E.preview = ($D.selected).find('#preview').eq(0);
-          $E.textarea.bind('input propertychange',$E.save);
-      }
+    // textarea
+    $E.textarea = ($D.selected).find('textarea').eq(0);
+    $E.textarea.bind('input propertychange',$E.save);
+    // preview
+    $E.preview = ($D.selected).find('#preview').eq(0);
   },
-
   // wraps currently selected text in textarea with strings a and b
   wrap: function(a,b,args) {
-    var isWiki = (window.location + '').includes('wiki');
-    if (!isWiki) this.refresh();
+    // this RegEx is: /wiki/ + any char not "/" + /edit
+    const isWikiCommentPage = (/\/wiki\/[^\/]+\/edit/).test(window.location.pathname);
+    // we don't need to refresh $E's values if we're on a page with a single comment form, ie. /wiki/new or /wiki/edit
+    if (window.location.pathname !== "/wiki/new" && !isWikiCommentPage && $D.selected) {
+      this.refresh();
+    }
     var len = $E.textarea.val().length;
```

Comment:
This is a pretty big change too. The `(window.location + '').includes('wiki')` part was originally done in #5650 to fix image upload in `/wiki/new` and `/wiki/{wiki name}/edit`... 

Unfortunately this had the side effect of running `refresh` on wiki comments pages (which `include('wiki')` in the `window.location`). I finally figured out this doesn't need to happen, because the `wiki#show` view just loads comments through `note#comments`. This used to make writing system tests more difficult because things ran differently for wiki comments.

The last note is that I'm thinking that `this.refresh();` may not need to be called at all. We could remove it entirely if we assign `$D.selected` in all the right places. But that's for another PR.